providers/common/provider_ctx.c

```diff
@@ -9,14 +9,31 @@
 
 #include <stdlib.h>
 #include "prov/provider_ctx.h"
+#include "prov/bio.h"
 
 PROV_CTX *PROV_CTX_new(void)
 {
-    return OPENSSL_zalloc(sizeof(PROV_CTX));
+    PROV_CTX *ctx = OPENSSL_zalloc(sizeof(*ctx));
+
+    if (ctx == NULL)
+        return NULL;
+
+# ifndef FIPS_MODULE
+    ctx->corebiometh = bio_prov_init_bio_method();
+    if (ctx->corebiometh == NULL) {
+        OPENSSL_free(ctx);
+        return NULL;
+    }
+# endif
+
+    return ctx;
 }
 
 void PROV_CTX_free(PROV_CTX *ctx)
 {
+# ifndef FIPS_MODULE
+    BIO_meth_free(ctx->corebiometh);
+# endif
```

Comment:
Likewise, I would leave this to the teardown functions

Reply:
I kind of feels wrong that PROV_CTX_free() does not know how to free its own internal members and it is the responsibility of something external to do that. However, for now, I've done it like this. I think we might want to revisit this in the future.